components/src/dynamo/profiler/thorough.py

```
@@ -400,6 +400,37 @@ async def run_thorough(
+    # Propagate device_type (e.g. "xpu") into all candidate DGD configs so
+    # that workers are deployed with the correct VLLM_TARGET_DEVICE env var.
+    device_type = getattr(ops, "device_type", "cuda")
+    if device_type != "cuda":
+        _modifier_cls = CONFIG_MODIFIERS.get(backend)
+        if _modifier_cls and hasattr(_modifier_cls, "set_device_type"):
+            for candidate in prefill_candidates:
+                try:
+                    candidate.dgd_config = _modifier_cls.set_device_type(
+                        candidate.dgd_config, device_type
+                    )
+                except Exception as _e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)